### PR TITLE
Open closed phases

### PR DESCRIPTION
**Pull request by masonlr**
This pull request represents work undertaken to support multiple challenge phases.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -7,6 +7,20 @@ services:
...
+      dockerfile: Dockerfile-open-loop
+    ports:
+      - 5001:5001      
+
+  nztc-closed-loop:
+    build:
+      context: environment
+      dockerfile: Dockerfile-closed-loop
+    ports:
+      - 5002:5002
```
Comment (masonlr) [inline docker-compose.yml:22]: These are isolated containers, so they should all use port 5000.

Actually, we'll only need a single `nztc` container and we'll select the relevant gym environment using a shell environment variable.

File: environment/Dockerfile-closed-loop [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.9-slim-buster
```
Comment (masonlr) [inline environment/Dockerfile-closed-loop:1]: Naming convention would be `Dockerfile.closed-loop`

File: environment/setup.py [hunk 3]
```diff
@@ -6,3 +6,15 @@
     version="0.1.0",
```
Comment (masonlr) [inline environment/setup.py:6]: Do we still need the `reference_environment_direct_deployment` env? Can we remove this?

File: evaluation/challenge_config.yaml [hunk 4]
```diff
@@ -57,6 +57,74 @@ challenge_phases:
...
+        type: checkbox
+        options: ["alpha", "beta", "gamma"]
+      - name: TrueFalseField
+        description: Sample
+        type: boolean
+        required: True
+    is_restricted_to_select_one_submission: False
+    is_partial_submission_evaluation_enabled: False
+
+challenge_phases:
```
Comment (masonlr) [inline evaluation/challenge_config.yaml:94]: This needs to be a YAML list. At the moment, the list is duplicated multiple times. There needs to be only a single `
challenge_phases` section.

File: environment/setup.py [hunk 5]
```diff
@@ -6,3 +6,15 @@
...
     install_requires=["gym", "matplotlib", "numpy", "pandas", "pycel"],
 )
+
+setup(
+    name="open_loop_env",
+    version="0.1.0",
+    install_requires=["gym", "matplotlib", "numpy", "pandas", "pycel"],
```
Comment (masonlr) [inline environment/setup.py:13]: Suggest locking the dependencies to specific versions.

**Comment by masonlr** [comment]
The reference to a custom folder `"./noise_sigma0.1_modified_workbook_finalized_env.py_DDPG/"` shouldn't be in the code: https://github.com/rangl-labs/netzerotc/blob/016432cfe794049c8f54241c1a417b1cd455108a/local_agent_training_and_evaluation/train.py#L13

**Comment by masonlr** [comment]
Have refactored so there is now a single `rangl` package with all three environments, you can now use:

```
gym.make("nztc-dev-v0")
gym.make("nztc-open-loop-v0")
gym.make("nztc-closed-loop-v0")
```

There is no need to duplicate any of the server logic as the agent can specify which gym environment it wants to use, for example with this code

https://github.com/rangl-labs/netzerotc/blob/f26b9bd0583c715bd20216df0a0f6d63fa19b3a0/random_agent_submission/agent.py#L11-L13

you can state the version of the environment that you want to use:

```python
# env_id = "nztc-dev-v0"
# env_id = "nztc-open-loop-v0"
env_id = "nztc-closed-loop-v0"
```

**Comment by masonlr** [comment]
Merging the YAML lists gives:

image

This way we can submit to multiple challenge phases.

**Comment by masonlr** [comment]
Can submit to multiple phases and view their respective leaderboards:


image

**Comment by moriartyjm** [comment]
> The reference to a custom folder `"./noise_sigma0.1_modified_workbook_finalized_env.py_DDPG/"` shouldn't be in the code:
> 
> https://github.com/rangl-labs/netzerotc/blob/016432cfe794049c8f54241c1a417b1cd455108a/local_agent_training_and_evaluation/train.py#L13

#100 fixes this

**Comment by masonlr** [comment]
Closing, as this has been replaced by https://github.com/rangl-labs/netzerotc/pull/101

**Pull request closed without merging**